### PR TITLE
The transport parameter removes EOED

### LINKED ISSUE #4475: Ben Kaduk's TLS Discuss 1
@kaduk said:

> (1) Rather a "discuss-discuss", but we seem to be requiring some changes
> to TLS 1.3 that are arguably out of charter.  In particular, in Section
> 8.3 we see that clients are forbidden from sending EndOfEarlyData and it
> (accordingly) does not appear in the handshake transcript.  The
> reasoning for this is fairly sound; we explicitly index our application
> data streams and any truncation will be filled in as a normal part of
> the recovery process, so the attack that EndOfEarlyData exists to
> prevent instrinsically cannot happen.  However, the only reason we'd be
> required to send it in the first place is if the server sends the
> "early_data" extension in EncryptedExtensions ... and we already have a
> bit of unpleasantness relating to the "early_data" extension, in that we
> have to use a sentinel value for max_early_data_size in NewSessionTicket
> to indicate that the ticket is good for 0-RTT, with the actual maximum
> amount of data allowed indicated elsewhere.  TLS extensions are cheap,
> so a new "quic_early_data" flag extension valid in CH, EE, and NST would
> keep us from conflating TLS and QUIC 0-RTT semantics, thus solving both
> problems at the same time.  On the other hand, that would be requiring
> implementations to churn just for process cleanliness, so we might also
> consider other alternatives, such as finessing the language and/or
> document metadata for how this specification uses TLS 1.3.
> (There are a couple other places in the COMMENT where we might suffer
> from scope creep regarding TLS behavior as well, but I did not mark them
> as DISCUSS since they are not changing existing specified behavior.)﻿

### PR DESCRIPTION
**Pull request by martinthomson**
As agreed via mail.

Closes #4475.

### CODE REVIEW

**Review by marten-seemann — APPROVED** [review]

**Review by janaiyengar — APPROVED** [review]

**Review by nibanks — APPROVED** [review]

**Review by seanturner — APPROVED** [review]

**Review by DavidSchinazi — APPROVED** [review]

**Review by ianswett — APPROVED** [review]

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -1821,6 +1821,9 @@ QUIC (such as the use of TLS with TCP defined in {{!TLS13}}).  A fatal
 unsupported_extension alert MUST be sent by an implementation that supports this
 extension if the extension is received when the transport is not QUIC.
 
+Negotiating the quic_transport_parameters extension causes the EndOfEarlyData to
```
Comment (kaduk) [inline draft-ietf-quic-tls.md:1824]:
```suggestion
Negotiating the quic_transport_parameters extension causes the requirement to send EndOfEarlyData to
```

**Review by kaduk — APPROVED** [review]
This gets the job done, with or without the suggested rewording.

File: draft-ietf-quic-tls.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-quic-tls.md:1824]: I think that the shorter phrasing is better.  It's not about not having a requirement to send EOED, but a requirement to not send EOED.  And, as this refers to the very next section, the fewer words expended here the better.

Comment (seanturner) [reply draft-ietf-quic-tls.md:1824]: For the question is whether an implementer (bot being on mind you) would be confused by the lack of "requirement to send" phrase, I tend to think that without it's still clear that you don't send EOED.

**Pull request merged**